apm-lambda-extension/logsapi/subscribe_test.go

```diff
@@ -1,136 +0,0 @@
-// Licensed to Elasticsearch B.V. under one or more contributor
-// license agreements. See the NOTICE file distributed with
-// this work for additional information regarding copyright
-// ownership. Elasticsearch B.V. licenses this file to you under
-// the Apache License, Version 2.0 (the "License"); you may
-// not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing,
-// software distributed under the License is distributed on an
-// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-// KIND, either express or implied.  See the License for the
-// specific language governing permissions and limitations
-// under the License.
-
-package logsapi
-
-import (
-	"bytes"
-	"context"
-	"encoding/json"
-	"net/http"
-	"net/http/httptest"
-	"testing"
-
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-)
-
-func TestSubscribeWithSamLocalEnv(t *testing.T) {
-	t.Setenv("AWS_SAM_LOCAL", "true")
-
-	_, err := Subscribe(context.Background(), "testID", []EventType{Platform})
-	assert.Error(t, err)
-}
-
-func TestSubscribeWithLambdaFunction(t *testing.T) {
-	t.Setenv("AWS_LAMBDA_FUNCTION_NAME", "mock")
-
-	_, err := Subscribe(context.Background(), "testID", []EventType{Platform})
-	assert.Error(t, err, "listen tcp: lookup sandbox: no such host")
-}
-
-func TestSubscribeAWSRequest(t *testing.T) {
```

Comment:
I don't see any equivalent of this test. Is that on purpose? Seems like a useful test, ensuring the low level log event delivery works.

I wonder if we should move the `ProcessLogs` and `ProcessPlatformReport` methods out of this package, and into either the `app` or `extension` package? Those methods are business logic, and I feel like it would be a better separation if we made `logsapi` just about delivering the raw log events. That would also make it more easily testable.

Reply:
Yep, the logs listener is not tested. I've added a new test to make sure it is handling requests properly.

> I wonder if we should move the ProcessLogs and ProcessPlatformReport methods out of this package, and into either the app or extension package? Those methods are business logic, and I feel like it would be a better separation if we made logsapi just about delivering the raw log events. That would also make it more easily testable.

I agree with you there, that's probably the last thing preventing a proper decoupling between logsapi and extension package. I think we should move those methods but I wanted to keep the focus of this PR to the `logsapi` package, without touching external code.

Let me know if you want to move them in this PR!

Reply:
OK, sounds good - I'm happy for that to be done in a followup.